### PR TITLE
Improve signature handling with DNF

### PR DESCRIPTION
**Pull request by ctron**

### CODE REVIEW

File: rpm/src/main/java/org/eclipse/packager/rpm/app/Dumper.java [hunk 1]
```diff
@@ -60,31 +65,46 @@ public static String dumpFlag ( final int value, final IntFunction<Optional<?>>
 
     public static void dumpAll ( final RpmInputStream in ) throws IOException
     {
-        final RpmLead lead = in.getLead ();
-        System.out.format ( "Version: %s.%s%n", lead.getMajor (), lead.getMinor () );
-        System.out.format ( "Name: %s%n", lead.getName () );
-        System.out.format ( "Signature Version: %s%n", lead.getSignatureVersion () );
-        System.out.format ( "Type: %s, Arch: %s, OS: %s%n", dumpFlag ( lead.getType (), Type::fromValue ), dumpFlag ( lead.getArchitecture (), Architecture::fromValue ), dumpFlag ( lead.getOperatingSystem (), OperatingSystem::fromValue ) );
+        final RpmLead lead = in.getLead();
```
Comment (dwalluck) [inline rpm/src/main/java/org/eclipse/packager/rpm/app/Dumper.java:68]: Small formatting issue.

File: rpm/src/main/java/org/eclipse/packager/rpm/build/PayloadProcessors.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+            }
+
+            @Override
+            public void finish ( final Header<RpmTag> header )
+            {
+                final String raw = Hex.toHexString ( digestRaw.digest () );
+                final String compressed = Hex.toHexString ( digestCompressed.digest () );
+                header.putStringArray ( RpmTag.PAYLOAD_DIGEST, compressed );
+                header.putStringArray ( RpmTag.PAYLOAD_DIGEST_ALT, raw );
+                header.putInt ( RpmTag.PAYLOAD_DIGEST_ALGO, algorithm.getTag() );
```
Comment (dwalluck) [inline rpm/src/main/java/org/eclipse/packager/rpm/build/PayloadProcessors.java:64]: Small formatting issue.

File: rpm/src/main/java/org/eclipse/packager/rpm/build/RpmBuilder.java [hunk 3]
```diff
@@ -961,6 +957,10 @@ private void fillHeader ()
         {
             this.header.putSize ( 0, RpmTag.SIZE, RpmTag.LONGSIZE );
         }
+
+        // add additional headers
+
+        this.header.putAll ( finished.getAdditionalHeader() );
```
Comment (dwalluck) [inline rpm/src/main/java/org/eclipse/packager/rpm/build/RpmBuilder.java:963]: Small formatting issue.

File: rpm/src/main/java/org/eclipse/packager/rpm/build/RpmBuilder.java [hunk 4]
```diff
@@ -1093,31 +1093,25 @@ public String getName ()
...
 
-        leadBuilder.fillFlagsFromHeader ( this.header, createLeadArchitectureMapper (), createLeadOperatingSystemMapper () );
-
-        if ( this.headerCustomizer != null )
-        {
-            this.headerCustomizer.accept ( this.header );
-        }
+            if ( this.headerCustomizer != null)
+            {
+                this.headerCustomizer.accept(this.header);
```
Comment (dwalluck) [inline rpm/src/main/java/org/eclipse/packager/rpm/build/RpmBuilder.java:1108]: Small formatting issue.

File: rpm/src/main/java/org/eclipse/packager/rpm/build/RpmBuilder.java [hunk 5]
```diff
@@ -1093,31 +1093,25 @@ public String getName ()
...
+            }
 
-        try ( final RpmWriter writer = new RpmWriter ( this.targetFile, leadBuilder, this.header, this.options.getHeaderCharset (), this.options.getOpenOptions () ) )
-        {
-            writer.addAllSignatureProcessors ( this.signatureProcessors );
-            writer.setPayload ( this.recorder );
+            try ( final RpmWriter writer = new RpmWriter ( this.targetFile, leadBuilder, this.header, this.options.getHeaderCharset (), this.options.getOpenOptions() ) ) {
```
Comment (dwalluck) [inline rpm/src/main/java/org/eclipse/packager/rpm/build/RpmBuilder.java:1111]: Small formatting issue.

File: rpm/src/main/java/org/eclipse/packager/rpm/signature/RsaSignatureProcessor.java [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+/**
+ * An RSA signature processor for both header and payload.
+ */
+public class RsaSignatureProcessor implements SignatureProcessor
+{
+    private final static Logger logger = LoggerFactory.getLogger ( RsaSignatureProcessor.class );
+
+    private final PGPSignatureGenerator signatureGenerator;
+
+    protected RsaSignatureProcessor(final PGPPrivateKey privateKey, final int hashAlgorithm )
```
Comment (dwalluck) [inline rpm/src/main/java/org/eclipse/packager/rpm/signature/RsaSignatureProcessor.java:39]: Small formatting issue.

File: rpm/src/main/java/org/eclipse/packager/rpm/signature/RsaSignatureProcessor.java [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+    {
+        Objects.requireNonNull ( privateKey );
+
+        final BcPGPContentSignerBuilder contentSignerBuilder = new BcPGPContentSignerBuilder ( privateKey.getPublicKeyPacket ().getAlgorithm (), hashAlgorithm);
```
Comment (dwalluck) [inline rpm/src/main/java/org/eclipse/packager/rpm/signature/RsaSignatureProcessor.java:43]: Small formatting issue.

File: rpm/src/main/java/org/eclipse/packager/rpm/signature/RsaSignatureProcessor.java [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+        this.signatureGenerator = new PGPSignatureGenerator ( contentSignerBuilder );
+
+        try {
+            this.signatureGenerator.init ( PGPSignature.BINARY_DOCUMENT, privateKey);
```
Comment (dwalluck) [inline rpm/src/main/java/org/eclipse/packager/rpm/signature/RsaSignatureProcessor.java:47]: Small formatting issue.

File: rpm/src/main/java/org/eclipse/packager/rpm/signature/RsaSignatureProcessor.java [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+    private final PGPSignatureGenerator signatureGenerator;
+
+    protected RsaSignatureProcessor(final PGPPrivateKey privateKey, final int hashAlgorithm )
+    {
+        Objects.requireNonNull ( privateKey );
+
+        final BcPGPContentSignerBuilder contentSignerBuilder = new BcPGPContentSignerBuilder ( privateKey.getPublicKeyPacket ().getAlgorithm (), hashAlgorithm);
+        this.signatureGenerator = new PGPSignatureGenerator ( contentSignerBuilder );
+
+        try {
```
Comment (dwalluck) [inline rpm/src/main/java/org/eclipse/packager/rpm/signature/RsaSignatureProcessor.java:46]: Small formatting issue. I think you have braces on their own lines normally.

**Review by dwalluck — CHANGES_REQUESTED** [review]
Looks good to me. I marked some formatting (whitespace) issues I saw, but they are not important.

**Comment by ctron** [comment]
> Looks good to me. I marked some formatting (whitespace) issues I saw, but they are not important.

Thanks for finding all of these :) I am struggling with the formatting. I think it is time to move to some more standard formatting, and let the IDE do it. I will finish this up and create a PR which changes the formatting of the whole repository. So we can judge, discuss, and maybe merge.

**Pull request merged**